src/formats.ts

```diff
@@ -169,6 +191,19 @@ function uri(str: string): boolean {
   return NOT_URI_FRAGMENT.test(str) && URI.test(str)
 }
 
+function validateInteger(bits: number): (value: number | string) => boolean {
+  const max = BigInt(2) ** BigInt(bits - 1)
+  const min = max * BigInt(-1)
+  return (value) => Number.isInteger(+value) && BigInt(value) <= max && BigInt(value) >= min
+}
+
+function validateNumber(bits: number): (value: number | string) => boolean {
+  const max = Number(BigInt(2) ** BigInt(bits - 1))
+  const min = max * -1
+
+  return (value) => max >= value && min <= value
```

Comment:
strictly ">" I think (although the limit is slightly lower I think)...

Reply:
This is where it becomes a bit of a challenge
Javascript integers go up to 2^53 hence the BigInt type
Ideally AJV would use BigInt for all its integer handling, however BigInt only plays nice with other BigInts :-(
(e.g. `2*BigInt(2)` fails `BigInt(2)*BigInt(2)` succeeds.
So either we stick to basic javascript integers and skip the whole BigInt stuff of we need to come up with something clever ..